### PR TITLE
[SPARK-21904] [SQL] Rename tempTables to tempViews in SessionCatalog

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?
`tempTables` is not right. To be consistent, we need to rename the internal variable names/comments to tempViews in SessionCatalog too.

### How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
@cloud-fan @sameeragarwal

**Comment by HyukjinKwon** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala [hunk 1]
```diff
@@ -90,13 +87,13 @@ class SessionCatalog(
       new SQLConf().copy(SQLConf.CASE_SENSITIVE -> true))
   }
 
-  /** List of temporary tables, mapping from table name to their logical plan. */
```
Comment (viirya) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala]: The code comment of `SessionCatalog` also includes wording `...it also manages temporary tables and ...`

**Comment by viirya** [comment]
LGTM

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 2]
```diff
@@ -812,7 +812,7 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
         sql("ALTER TABLE tab1 RENAME TO default.tab2")
       }
       assert(e.getMessage.contains(
-        "RENAME TEMPORARY TABLE from '`tab1`' to '`default`.`tab2`': " +
+        "RENAME TEMPORARY VIEW from '`tab1`' to '`default`.`tab2`': " +
```
Comment (viirya) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: We can rename this test name from `rename temporary table...` to ``rename temporary view...`.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 3]
```diff
@@ -860,7 +860,7 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
         sql("ALTER TABLE tab1 RENAME TO tab2")
       }
       assert(e.getMessage.contains(
-        "RENAME TEMPORARY TABLE from '`tab1`' to '`tab2`': destination table already exists"))
+        "RENAME TEMPORARY VIEW from '`tab1`' to '`tab2`': destination table already exists"))
```
Comment (viirya) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: ditto. This test name is also `rename temporary table...`

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 4]
```diff
@@ -812,7 +812,7 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
         sql("ALTER TABLE tab1 RENAME TO default.tab2")
       }
       assert(e.getMessage.contains(
-        "RENAME TEMPORARY TABLE from '`tab1`' to '`default`.`tab2`': " +
+        "RENAME TEMPORARY VIEW from '`tab1`' to '`default`.`tab2`': " +
           "cannot specify database name 'default' in the destination table"))
 
       val catalog = spark.sessionState.catalog
```
Comment (viirya) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: At L823 there is a test named `rename temporary table` too.

**Comment by felixcheung** [comment]
try rebasing? it should be fixed by this commit https://github.com/apache/spark/commit/36b48ee6e92661645648a001d0d83623a8e5d601

**Review by felixcheung — APPROVED** [review]

**Comment by gatorsmile** [comment]
Thanks! Merged to master.

**Pull request closed without merging**